mesonbuild/options.py

```diff
@@ -1370,16 +1370,28 @@ def initialize_from_top_level_project_call(self,
                 if proj_key in self.options:
                     self.set_option(proj_key, valstr, True)
                 else:
-                    # Fail on unknown options that we can know must
-                    # exist at this point in time. Subproject and compiler
-                    # options are resolved later.
-                    #
-                    # Some base options (sanitizers etc) might get added later.
-                    # Permitting them all is not strictly correct.
-                    if key.subproject is None and not self.is_compiler_option(key) and not self.is_base_option(key):
-                        raise MesonException(f'Unknown options: "{keystr}"')
                     self.pending_options[key] = valstr
 
+    def validate_cmd_line_options(self, cmd_line_options: OptionStringLikeDict) -> None:
+        unknown_options = []
+        for keystr, valstr in cmd_line_options.items():
+            if isinstance(keystr, str):
+                key = OptionKey.from_string(keystr)
+            else:
+                key = keystr
+            # Fail on unknown options that we can know must exist at this point in time.
+            # Subproject and compiler options are resolved later.
+            #
+            # Some base options (sanitizers etc) might get added later.
+            # Permitting them all is not strictly correct.
+            if key.subproject is None and not self.is_compiler_option(key) and not self.is_base_option(key) and \
+               key in self.pending_options:
+                unknown_options.append(str(key))
+
+        if unknown_options:
+            keys = ', '.join(unknown_options)
+            raise MesonException(f'Unknown options: "{keys}"')
```

Comment:
Quotes maybe should technically go around each item, not around the whole list?

Reply:
Sure.